example/image-classification/predict-cpp/CMakeLists.txt

```diff
@@ -5,11 +5,11 @@ OpenCV version >= 3 should be enabled and found to build image classification ex
   return()
 endif()
 
-if(NOT MSVC)
+if(NOT MSVC AND NOT CMAKE_BUILD_SHARED)
   set(IMG_CLASSIFICATION_EXAMPLE_STATIC_LINK ON CACHE BOOL "\
 Link mxnet library statically in the c++ image classification example")
 else()
-  # disable static linking on Windows
+  # disable static linking on Windows & CMAKE_BUILD_SHARED
   set(IMG_CLASSIFICATION_EXAMPLE_STATIC_LINK OFF)
 endif()
```

Comment:
Let's delete this too and let cmake handle things automatically based on `CMAKE_BUILD_SHARED`